Modix/Modules/BetModule.cs

```diff
@@ -0,0 +1,230 @@
+        private const int StageStepLengthInSeconds = 5;
+        private const int RollingDurationInSeconds = 30;
+        private const int RollingStepLengthInSeconds = 5;
+
+        private static int _stageCounter;
+        private static int _rollingCounter;
+        private static IUser _sessionOwner;
+        private static BettingState _state;
+        private static Timer _stageTimer;
+        private static Timer _rollingTimer;
+        private static List<IUser> _joinedUsers = new List<IUser>();
+        private static Dictionary<IUser, int> _rolls = new Dictionary<IUser, int>();
+        private static Random _randomGenerator = new Random();
+
+        [Command("Stage"), Summary("Sets the stage for a new betting game. Players can join during this time with !bet join")]
+        public async Task StageAsync()
+        {
+            if (_state != BettingState.Idle)
+            {
+                if (_state == BettingState.Stage && Context.User.Id != _sessionOwner.Id)
+                {
+                    await Context.Channel.SendMessageAsync($"MODiX is currently setting the stage for a bet. There is only `{StageDurationInSeconds - _stageCounter}` seconds left to join. `!bet join` to participate in this session");
+                }
+                else if (_state == BettingState.Stage && Context.User.Id == _sessionOwner.Id)
+                {
+                    await Context.Channel.SendMessageAsync($"You're the owner of the current session, {Context.User.Mention}. Type `!bet start` to begin the game!");
+                }
+                else
+                {
+                    await Context.Channel.SendMessageAsync("MODiX is currently in a bet. Please wait for the current game to end");
+                }
+            }
+
+            _state = BettingState.Stage;
+            _sessionOwner = Context.User;
+            _joinedUsers.Add(_sessionOwner);
+            await Context.Channel.SendMessageAsync($"{Context.User.Mention} has set the stage for a new betting game. `!bet join` to participate in this session. `!bet start` to begin!");
+
+            _stageTimer = new Timer(async (state) =>
+            {
+                if (_state == BettingState.Rolling)
+                {
+                    // It's possible there could be a race condition. Account for it here.
+                    ResetStageTimer();
+                }
+                else if (_stageCounter >= StageDurationInSeconds && _joinedUsers.Count < 2)
+                {
+                    await Context.Channel.SendMessageAsync($"Not enough players joined {Context.User.Mention}'s session. Start another game with `!bet stage`");
+                    ResetStageTimer();
+                }
+
+                _stageCounter += StageStepLengthInSeconds;
+
+            }, _stageCounter, 0, 1000 * StageStepLengthInSeconds);
+        }
+
+        [Command("Start"), Summary("Starts a new session of the game")]
+        public async Task StartAsync()
```

Comment:
I feel like a jerk for just saying this is too long but.. this is too long. Can we maybe extract the lambda on line 96 into an own method or something? Because that thing is big enough